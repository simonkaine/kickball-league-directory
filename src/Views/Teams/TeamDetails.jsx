import { useEffect, useState} from 'react';
import { getTeamById } from '../../Services/teams';
// import { useParams } from 'react-router';

export default function TeamDetails({ match }) {
    const { teamId } = match.params;
    const [team, setTeam] = useState(null);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        getTeamById(teamId).then((response) => setTeam(response))
        setTimeout(() => {
            setLoading(false)
        }, 600)
    }, [teamId]);

    if (loading) return <h1 style={{height: '100vh', fontSize: '3em', marginTop: '350px'}}>...Loading team</h1>;

    return (
        <div className='TeamDetails' style={{border: '1px solid black', padding: '100px 0 0 0'}}>
            <h1 style={{fontFamily: 'Century Gothic', fontSize: '4em'}}>
                {team.name}
            </h1>
            <h3 style={{fontFamily: 'Century Gothic', fontSize: '2em'}}>
                From: {team.city}, {team.state}
            </h3>

            <ul aria-label='team players' style={{listStyleType: 'none', height: '100vh'}}>
                {team.players.map((player) => {
                    return (
                        <li key={player.id} style={{textDecoration: 'underline', padding: '10px'}}>
                            Position: {player.position}
                                {' '}--{' '}
                            Name: {player.name}
                        </li>
                    );
                })}
            </ul>
        </div>
    );
}
