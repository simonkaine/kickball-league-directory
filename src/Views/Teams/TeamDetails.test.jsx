import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import TeamDetails from "./TeamDetails";

it('Should test for detailed view of team', async () => {
    const {container} = render(
        <MemoryRouter>
            <TeamDetails match={{ params: {teamId: '1'} }}/>
        </MemoryRouter> 
    );
    
    screen.getByText('...Loading team');
    const theTeamName = await screen.findByText(/Bridge City Sneakers/i);
    expect(theTeamName).toBeInTheDocument()
    expect(container).toMatchSnapshot();
});