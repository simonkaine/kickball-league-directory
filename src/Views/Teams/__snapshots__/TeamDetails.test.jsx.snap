// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test for detailed view of team 1`] = `
<div>
  <div
    class="TeamDetails"
    style="border: 1px solid black; padding: 100px 0px 0px 0px;"
  >
    <h1
      style="font-family: Century Gothic; font-size: 4em;"
    >
      Bridge City Sneakers
    </h1>
    <h3
      style="font-family: Century Gothic; font-size: 2em;"
    >
      From: 
      Portland
      , 
      OR
    </h3>
    <ul
      aria-label="team players"
      style="list-style-type: none; height: 100vh;"
    >
      <li
        style="text-decoration: underline; padding: 10px;"
      >
        Position: 
        Pitcher
         
        --
         
        Name: 
        Ben E. Jetts
      </li>
      <li
        style="text-decoration: underline; padding: 10px;"
      >
        Position: 
        Catcher
         
        --
         
        Name: 
        Hans Olo
      </li>
      <li
        style="text-decoration: underline; padding: 10px;"
      >
        Position: 
        Outfield
         
        --
         
        Name: 
        Ron Long
      </li>
      <li
        style="text-decoration: underline; padding: 10px;"
      >
        Position: 
        Shortstop
         
        --
         
        Name: 
        Shorty Biggum
      </li>
    </ul>
  </div>
</div>
`;
